### PR TITLE
server: differentiate email for subscription cancelation

### PR DESCRIPTION
**Pull request by codenoid**
On the existing code, Polar send the subscription created email for when the even is subscription cancelation.

### CODE REVIEW

**Comment by codenoid** [comment]
Sorry for the gipiti comment, 3AM Pull Request kinda hit different

**Comment by frankie567** [comment]
Hey @codenoid 👋 

Thank you for this! So, you mean that a confirmation email is sent **after cancelling** a subscription plan? Did you experience this? That sounds really unexpected 😅

**Comment by codenoid** [comment]
@frankie567 correct, that was what happen when I testing (in polar.so) to cancel my subscription

**Comment by frankie567** [comment]
Thank you @codenoid, I've checked on my side and indeed experienced the same behavior; really appreciate your report on this!

I'll have a look at the PR ;)

File: server/polar/subscription/service.py [hunk 1]
```diff
@@ -587,6 +588,41 @@ async def send_confirmation_email(
             from_email_addr="noreply@notifications.polar.sh",
         )
 
+    async def send_cancellation_email(
```
Comment (frankie567) [inline server/polar/subscription/service.py:593]: Nice, but this isn't called anywhere, right?

Maybe we should add in `update_subscription_from_stripe` above:

```py
elif subscription.cancel_at_period_end:
    await self.send_cancellation_email(session, subscription)
```

File: server/polar/subscription/service.py [hunk 2]
```diff
@@ -506,6 +506,7 @@ async def update_subscription_from_stripe(
             and subscription.started_at is not None
             and subscription.started_at.date()
             == subscription.current_period_start.date()
+            and not subscription.cancel_at_period_end
```
Comment (frankie567) [inline server/polar/subscription/service.py:509]: Good catch!

**Review by frankie567 — CHANGES_REQUESTED** [review]

File: server/polar/subscription/service.py (hunk 1 above)
Comment (codenoid) [reply server/polar/subscription/service.py:593]: ah, rookie mistake, code updated!

**Comment by frankie567** [comment]
Thank you for this @codenoid, that's really helpful 🙏

**Comment by frankie567** [comment]
`date` is not a builtin filter in Jinja2 🙂 Fixed and added a unit test here: #3948

**Pull request merged**